src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -156,6 +238,48 @@ class OwlViTObjectDetectionOutput(ModelOutput):
     class_embeds: torch.FloatTensor = None
     text_model_last_hidden_state: Optional[torch.FloatTensor] = None
     vision_model_last_hidden_state: Optional[torch.FloatTensor] = None
+    text_model_output: BaseModelOutputWithPooling = None
+    vision_model_output: BaseModelOutputWithPooling = None
+
+
+@dataclass
+class OwlViTImageGuidedObjectDetectionOutput(ModelOutput):
+    """
+    Args:
+    Output type of [`OwlViTForObjectDetection.image_guided_detection`].
+        logits (`torch.FloatTensor` of shape `(batch_size, num_patches, num_queries)`):
+            Classification logits (including no-object) for all queries.
+        target_pred_boxes (`torch.FloatTensor` of shape `(batch_size, num_patches, 4)`):
+            Normalized boxes coordinates for all queries, represented as (center_x, center_y, width, height). These
+            values are normalized in [0, 1], relative to the size of each individual target image in the batch
+            (disregarding possible padding). You can use [`~OwlViTFeatureExtractor.post_process`] to retrieve the
+            unnormalized bounding boxes.
+        query_pred_boxes (`torch.FloatTensor` of shape `(batch_size, num_patches, 4)`):
+            Normalized boxes coordinates for all queries, represented as (center_x, center_y, width, height). These
+            values are normalized in [0, 1], relative to the size of each individual query image in the batch
+            (disregarding possible padding). You can use [`~OwlViTFeatureExtractor.post_process`] to retrieve the
+            unnormalized bounding boxes.
+        image_embeds (`torch.FloatTensor` of shape `(batch_size, patch_size, patch_size, output_dim`):
+            Pooled output of [`OwlViTVisionModel`]. OWL-ViT represents images as a set of image patches and computes
+            image embeddings for each patch.
+        query_image_embeds (`torch.FloatTensor` of shape `(batch_size, patch_size, patch_size, output_dim`):
+            Pooled output of [`OwlViTVisionModel`]. OWL-ViT represents images as a set of image patches and computes
+            image embeddings for each patch.
+        class_embeds (`torch.FloatTensor` of shape `(batch_size, num_patches, hidden_size)`):
+            Class embeddings of all image patches. OWL-ViT represents images as a set of image patches where the total
+            number of patches is (image_size / patch_size)**2.
+        vision_model_last_hidden_state (`torch.FloatTensor` of shape `(batch_size, num_patches + 1, hidden_size)`)):
```

Comment:
There's no need for this argument, instead one should include `vision_model_output` and `text_model_output` here, from which the last hidden states can be obtained.

Similarly, the image_embeds and text_embeds can be included here (I don't see any text_embeds here?), which, similar to CLIP contain the projected last hidden states.

Reply:
I will replace this with the `vision_model_output `but the image_guided_detection method takes no text input